### PR TITLE
Add Cloudfront details to website show GET endpoint

### PR DESCRIPTION
**Pull request by fishnix**

### CODE REVIEW

File: cloudfront/distribution.go [hunk 1]
```diff
@@ -49,3 +50,36 @@ func (c *CloudFront) ListDistributions(ctx context.Context) ([]*cloudfront.Distr
...
+
+// GetDistributionByName gets a cloudfront distribution by the name (by searching until it finds the matching alias)
+func (c *CloudFront) GetDistributionByName(ctx context.Context, name string) (*cloudfront.DistributionSummary, error) {
+	log.Infof("searching for cloudfront distribution %s", name)
+
+	input := &cloudfront.ListDistributionsInput{MaxItems: aws.Int64(100)}
+
+	var distribution *cloudfront.DistributionSummary
+	err := c.Service.ListDistributionsPagesWithContext(ctx, input,
+		func(out *cloudfront.ListDistributionsOutput, lastPage bool) bool {
```
Comment (tenyo) [inline cloudfront/distribution.go]: Pretty cool 👍 I didn't realize it's so tricky to get information about a distribution but I guess all the functions use the `id` and we do it based on the `alias` name

File: route53/records.go [hunk 2]
```diff
@@ -63,6 +63,43 @@ func (r *Route53) GetRecord(ctx context.Context, zoneID, host, recordType string
 	return nil, apierror.New(apierror.ErrNotFound, msg, nil)
 }
 
+// GetRecordByName gets a route53 resource record by name and by type if one is specified
+func (r *Route53) GetRecordByName(ctx context.Context, zoneID, name, recordType string) (*route53.ResourceRecordSet, error) {
```
Comment (tenyo) [inline route53/records.go]: This function seems to be doing the same thing as `GetRecord` above (but more efficient). Do we need to keep both?

File: route53/records_test.go [hunk 3]
```diff
@@ -407,4 +425,109 @@ func TestGetRecord(t *testing.T) {
...
+		Service: newmockRoute53Client(t, nil),
+		Domains: map[string]common.Domain{
+			"hyper.converged": common.Domain{
+				CertArn: "arn:aws:acm::12345678910:certificate/111111111-2222-3333-4444-555555555555",
+			},
+		},
+	}
+
+	expected := &testResourceRecordSet
+	out, err := r.GetRecordByName(context.TODO(), testHostedZoneID, "foobar.hyper.converged", "A")
```
Comment (tenyo) [inline route53/records_test.go]: Do we also need to test for success if the name ends in a `.`

File: route53/records.go (hunk 2 above)
Comment (fishnix) [reply route53/records.go]: true, good catch 💀 code

**Comment by tenyo** [comment]
👍

**Comment by damnski** [comment]
lots of tests :+1:

**Pull request merged**